repair/row_level.cc

```diff
@@ -439,6 +439,11 @@ using is_local_reader = bool_class<class is_local_reader_tag>;
 
 };
 
+struct repair_writer_stats {
+    uint64_t nr_parttion_start = 0;
+    uint64_t nr_parttion_end = 0;
+};
+
 class repair_writer {
```

Comment:
I don't understand this. Why do we need to count and compare starts and ends? If we abort something, we should tear down the stream, not try to clean it up. If the other side is dead, it won't respond to additional writes, and the socket buffer may be full.

Reply:
This has nothing to do with streams. partition_end is not synced between nodes on wire. We generate partition_end on the node that needs to write the mutation fragments. partition_end is implied when a new partition_start is received.

Reply:
when no more fragments from the peer, i.e., the repair is done, we write a partition_end in repair_row_level_stop.

Reply:
I still don't understand it.

There is a stream of fragments from one end of the queue to the other, no? Why do you need to count them?

Reply:
partition_end is not transferred between nodes